ui/app/AppLayouts/Onboarding/views/ProfileFetchingView.qml

```diff
@@ -59,56 +55,167 @@ Item {
+                    to: model.totalMessages
+                    value: model.loadedMessages
+                    backgroundColor: Theme.palette.baseColor5
+                    backgroundBorderColor: backgroundColor
+                    fillColor: {
+                        if (model.totalMessages > 0 && model.totalMessages === model.loadedMessages)
+                            return Theme.palette.successColor1
+                        return Theme.palette.primaryColor1
+                    }
+                }
             }
+
+            ScrollBar.vertical: ScrollBar {}
         }
 
         StatusButton {
             id: button
-            visible: false
             Layout.alignment: Qt.AlignHCenter
+            Layout.preferredWidth: root.startupStore.currentStartupState.stateType === Constants.startupState.profileFetching? 80 : -1
             focus: true
-            text: Constants.onboarding.profileFetching.tryAgainText
-            onClicked: {
-                root.startupStore.doPrimaryAction()
-            }
-        }
+            enabled: root.startupStore.currentStartupState.stateType !== Constants.startupState.profileFetching
 
-        StatusBaseText {
-            id: link
-            visible: false
-            Layout.alignment: Qt.AlignHCenter
-            font.pixelSize: Constants.keycard.general.buttonFontSize
-            text: Constants.onboarding.profileFetching.createNewProfileText
-            color: Theme.palette.primaryColor1
-            font.underline: linkMouseArea.containsMouse
-            MouseArea {
-                id: linkMouseArea
-                anchors.fill: parent
-                cursorShape: Qt.PointingHandCursor
-                hoverEnabled: true
-                onClicked: {
-                    root.startupStore.doSecondaryAction()
+            Timer {
+                id: timer
+                interval: 1000
+                running: root.startupStore.currentStartupState.stateType === Constants.startupState.profileFetching
+                repeat: true
+                onTriggered: {
+                    d.counter--
+                    let timerVal = d.getCurrTimerValue()
+                    if (timerVal === "0:00") {
+                        root.startupStore.doPrimaryAction()
+                    }
                 }
             }
+
+            onClicked: {
+                root.startupStore.doPrimaryAction()
+            }
         }
 
         Item {
             Layout.fillWidth: true
             Layout.fillHeight: true
         }
     }
+
+    states: [
+        State {
+            name: Constants.startupState.profileFetching
+            when: root.startupStore.currentStartupState.stateType === Constants.startupState.profileFetching
+            PropertyChanges {
+                target: title
+                text: qsTr("Fetching data...")
+            }
+            PropertyChanges {
+                target: button
+                text: d.getCurrTimerValue()
```

Comment:
function getCurrTimerValue {..} could be reduced to

`text: Qt.formatTime(new Date(0,0,0,0,0,d.counter), "mm:ss")`